src/storage/core.rs

```diff
@@ -486,26 +504,43 @@ where
             .map(|blob| blob.data.read_any(key, meta, false))
             .collect();
         debug!("read with optional meta {} closed blobs", stream.len());
-        let mut task = stream.skip_while(Result::is_err);
-        task.next().await.ok_or_else(Error::not_found)?
+        let mut task = stream.skip_while(|r| {
+            if let Ok(h) = r {
+                !h.is_presented()
+            } else {
+                true
+            }
+        });
+        task.next().await.unwrap_or(Ok(ReadResult::NotFound))
     }
 
     #[allow(dead_code)]
-    async fn get_data_any(safe: &Safe<K>, key: &K, meta: Option<&Meta>) -> Result<Vec<u8>> {
+    async fn get_data_any(
+        safe: &Safe<K>,
+        key: &K,
+        meta: Option<&Meta>,
+    ) -> Result<ReadResult<Vec<u8>>> {
         let blobs = safe.blobs.read().await;
         let stream: FuturesUnordered<_> = blobs
```

Comment:
Unfortunately, we should use `FuturesOrdered<_>`. Otherwise the deletion record can be skipped